### DISCUSSION TITLE
send message with multiple attachment

### DISCUSSION THREAD

**Discussion by junaidmgithub** [AsamK/signal-cli#665] [Q&A]
i am having issue with send message with multiple attachment.
only first attachment is taken. rest not coming to signal

**Comment by babbisch** [comment]
How do you set the command?

Try
`send RECEPIENT -a /path/to/image1.png /path/to/images2.png`
worked for me

**Comment by johnfreed** [comment]
I am having the same issue. It doesn't always choose the first attachment; when I send three attachments, and vary the order, it always sends the same one attachment. I have done some debugging to determine that everything is working well (all three attachments make it to sendMessage() in Manager.java. Unfortunately I don't have the skills to determine what happens after that. 

In AttachmentUtils.java, there is a createAttachment() method that uses a nonstandard (deprecated?) protocol to send to the Signal servers. It uses a resumableUploadSpec, for example, which exists nowhere in the current API so far as I can tell. So at that point I'm stumped. 

I'm using signal-cli on Fedora 34, openjdk-11.0.9-5

**Comment by johnfreed** [comment]
More information: The same thing happens when I use the "dbus-send" (raw) command:

dbus-send --session --type=method_call --print-reply --dest="org.asamk.Signal" /org/asamk/Signal org.asamk.Signal.sendMessage string:'Here is a message' array:string:'file1.pdf','file2.jpeg','file3.png' string:'+33000000000'

**Reply by johnfreed** [reply]
I'm wondering if the problem might be at the smartphone client. I'm using the official Signal app for Android.

**Reply by AsamK** [reply]
That could be it, as far as I know, the official clients only support multiple image attachments. Multiple attachments of different types are not supported

**Reply by johnfreed** [reply]
Some more evidence to support the idea that it's a Signal smartphone problem. 

I sent 3 PNG files, all were received fine.

Sent 2 PNG and 1 JPG, all were received fine.

Sent 2 PNG and 1 PDF file, only the PDF was received.

**Comment by johnfreed** [comment]
It appears that Signal processes images differently from other attachments. Specifically, for an image, the name is removed and replaced by an arbitrary string of characters, with no indication of MIME type or former file extension (.JPG, .PNG, etc.)

Other attachments, however, are sent along with their MIME type and original filename intact. It seems as if you can't mix and match these in the same send command.
